flow/set.go

```diff
@@ -252,20 +247,47 @@ func (fs *FlowSet) Dedup(field string) error {
 		return nil
 	}
 
-	visited := make(map[interface{}]bool)
+	dedupFlows := NewDedup(&deduped, field)
+	if err := dedupFlows.Flows(fs.Flows); err != nil {
+		return err
+	}
+	fs.Flows = deduped
+
+	return nil
+}
+
+// Group return only the flows that have a matching field in a FlowSet
+func (fs *FlowSet) Group(field string) error {
```

Comment:
Looks like more a "Sort" than a "Group". I would rather expect an output like this, even if I don't have a strong opinion here.

```
<GroupBy ID1>: [
 Flow{}
 Flow{}
] ,
<GroupBy ID2>: [
 Flow{}
 Flow{}
],
```

Reply:
Group is looks like a Sort but we don't report single/unique flow, as we only want flows that have a common/same matching field
Think about group like a union, but union is not the good word in gremlin syntax

Report the key (grouped by) like Sockets, have no opinion too on that, was just implementing as simple I can for the gremlin latency query

Reply:
futhermore, that why it's only applied after merge

Reply:
we could Group() do output format as you describe (with unqiue flow) and a Group(morethan=1) report that remove unique flow, What do you think ?

Reply:
mmm, I think "Group" means group not filtering. I would just do the group stuff and introduce a filter later based on the result of group.